ksqldb-serde/src/main/java/io/confluent/ksql/serde/protobuf/ProtobufSerdeFactory.java

```diff
@@ -145,18 +158,12 @@ private <T> KsqlConnectDeserializer<T> createDeserializer(
 
   private DataTranslator getDataTranslator(final Schema schema,
       final Optional<Schema> physicalSchema, final boolean isDeserializer) {
-    // If physical schema exists, we use physical schema to translate to connect data. During
-    // deserialization, if physical schema exists, we use original schema to translate to ksql data.
-    return physicalSchema.<DataTranslator>map(
-            value -> isDeserializer ? new ConnectDataTranslator(schema)
-                : new ConnectSRSchemaDataTranslator(value))
-        .orElseGet(() -> {
-          if (fullSchemaName.isPresent()) {
-            return new ProtobufDataTranslator(schema, fullSchemaName.get());
-          } else {
-            return new ConnectDataTranslator(schema);
-          }
-        });
+    // During deserialization, we use the original schema to translate to KSQL data.
+    if (isDeserializer || !physicalSchema.isPresent()) {
```

Comment:
physicalSchema will always exist?

Reply:
Yes, but I noticed there are places this does not exist yet. I will fix this after the regression issue  is fixed.